logger/drain/simple/drain.go

```diff
@@ -0,0 +1,117 @@
+package simple
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+)
+
+// This determines how often we'll replace an existing connection with a new one.  This ensures we
+// redial, which in turn will ensure any DNS changes affecting where logs are drained to don't
+// go unnoticed for too long.
+const connRefreshInterval = 5 * time.Minute
```

Comment:
Harder than it sounds.  Currently the configurer just watches `/deis/logs/drain` for changes.  When changes are noticed, that one value is passed to the factory to get the right drain back.  Making all those things configurable means watching more things _and_ expanding the interface for the factory so values for each of those things can be passed in-- which might not be entirely ideal since some of those things may not be relevant to other drain types.

More configuration options are nice, but in this case, I think the hard-coded values are pretty sensible and should suffice near term.  imo, we should re-evaluate this only if the community starts calling for it.